src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1349,6 +1349,16 @@ impl Decimal {
             return Res::Ok(self);
         }
 
+        let upper = (DIGITS_PER_WORD * word_buf_len * 2) as isize;
```

Comment:
It's just a safety value,  we can make sure that `shift` bigger than `DIGITS_PER_WORD * word_buf_len*2` will return an overflow result and smaller than `-DIGITS_PER_WORD * word_buf_len*2` will return a truncated result. And for `shift` in [-2*DIGITS_PER_WORD * word_buf_len, DIGITS_PER_WORD * word_buf_len * 2] which we can not sure if it is overflow or truncated, it will be checked below.  And we check the `shift` here just in case for panic caused by overflow(such as new cases in the test.) @breeswish